tests/wasmfs/wasmfs_getdents.c

```diff
@@ -18,35 +19,62 @@
 #include <sys/syscall.h>
 #include <unistd.h>
 
-void print(int fd) {
-  while (true) {
-    struct dirent d;
-    int nread = getdents(fd, &d, sizeof(d));
-    assert(nread != -1);
-    if (nread == 0) {
-      break;
-    }
-    printf("d.d_name = %s\n", d.d_name);
-    printf("d.d_off = %lld\n", d.d_off);
-    printf("d.d_reclen = %hu\n", d.d_reclen);
+void print_one(int fd) {
+  struct dirent d;
+  int nread = getdents(fd, &d, sizeof(d));
+  assert(nread != -1);
+  if (nread == 0) {
+    return;
+  }
+  printf("d.d_name = %s\n", d.d_name);
+  printf("d.d_reclen = %hu\n", d.d_reclen);
+  printf("d.d_type = %s\n\n",
+         (d.d_type == DT_REG)   ? "regular"
+         : (d.d_type == DT_DIR) ? "directory"
+                                : "???");
+}
+
+void print(const char* dir) {
+  struct dirent** entries;
+  int nentries = scandir(dir, &entries, NULL, alphasort);
+  assert(nentries != -1);
+  for (int i = 0; i < nentries; i++) {
+    printf("d.d_name = %s\n", entries[i]->d_name);
+    printf("d.d_reclen = %hu\n", entries[i]->d_reclen);
     printf("d.d_type = %s\n\n",
-           (d.d_type == DT_REG)   ? "regular"
-           : (d.d_type == DT_DIR) ? "directory"
-                                  : "???");
+           (entries[i]->d_type == DT_REG)   ? "regular"
+           : (entries[i]->d_type == DT_DIR) ? "directory"
+                                            : "???");
+    free(entries[i]);
   }
+  free(entries);
 }
 
 int main() {
+  // Set up the test root with the given backend.
+#ifdef MEMORY
+  backend_t backend = wasmfs_get_backend_by_path("/");
```

Comment:
This can just be set to NULL, in which case the parent backend will be used.
```suggestion
  backend_t backend = NULL;
```